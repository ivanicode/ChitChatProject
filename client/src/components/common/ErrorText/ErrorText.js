import React from 'react';



function ErrorText(props) {
    const defaultErrorText = 'Has≈Ço jest niezgodne';
    const errorText = props.error ? props.error : defaultErrorText;

    return (
    <span className="error active" aria-live="polite">{errorText}</span>
    )
}

export default ErrorText;