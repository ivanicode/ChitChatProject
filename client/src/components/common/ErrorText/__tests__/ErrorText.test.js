import React from 'react';
import {shallow} from 'enzyme';
import {ErrorText} from '../ErrorText';


describe('ErrorText component', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<ErrorText />)
        expect(wrapper).toMatchSnapshot()
    })
    it('should match snapshot with passed data', () => {
        const wrapper = shallow(<ErrorText error="Błąd"/>)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('span').text()).toEqual("Błąd")
        console.log(wrapper.find('span').text())
    })    
})