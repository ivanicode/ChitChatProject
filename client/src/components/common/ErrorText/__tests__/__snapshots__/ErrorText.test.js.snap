// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorText component should match snapshot 1`] = `
<span
  aria-live="polite"
  className="error active"
>
  Hasło jest niezgodne
</span>
`;

exports[`ErrorText component should match snapshot with passed data 1`] = `
<span
  aria-live="polite"
  className="error active"
>
  Błąd
</span>
`;
